org.uqbar.project.wollok.ui/src/org/uqbar/project/wollok/ui/quickfix/WollokDslQuickfixProvider.xtend

```diff
@@ -106,6 +109,47 @@ class WollokDslQuickfixProvider extends DefaultQuickfixProvider {
 		]
 	}
 
+	@Fix(WRONG_NUMBER_ARGUMENTS_CONSTRUCTOR_CALL)
+	def createConstructorInClass(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, Messages.WollokDslQuickFixProvider_create_constructor_class_name,
+			Messages.WollokDslQuickFixProvider_create_constructor_class_description, null) [ e, it |
+			val call = e as WConstructorCall
+			val clazz = call.classRef
+			val constructor = call.defaultStubConstructor.toString 
+			clazz.insertConstructor(constructor, it)
+		]
+	}
+
+	@Fix(WRONG_NUMBER_ARGUMENTS_CONSTRUCTOR_CALL)
+	def adjustConstructorCall(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, Messages.WollokDslQuickFixProvider_adjust_constructor_call_name,
+			Messages.WollokDslQuickFixProvider_adjust_constructor_call_description, null) [ e, it |
+			val call = e as WConstructorCall
+			val clazz = call.classRef
+			val numberOfParameters = call.numberOfParameters
+			val constructors = clazz.constructors.sortBy [ a | (a.parameters.size - numberOfParameters).abs ]
+			var paramsSize = 0
+			var WConstructor constructor = null
+			if (!constructors.isEmpty) {
+				constructor = constructors.head
+				paramsSize = constructor.parameters.size
+			}
+			val diffSize = numberOfParameters - paramsSize
+			var List<String> newParams = newArrayList
+			val List<String> paramsConstructor = if (constructor === null) newArrayList else constructor.parameters.map [ name ]
+			if (diffSize < 0) {
+				newParams = (call.arguments.map [ node.text.trim ]
```

Comment:
I do not recall seeing a test for this case, is there one?

Reply:
In ConstructorsQuickFixTest you have five tests:

- adjustConstructorCallParams1
- adjustConstructorCallParams2
- adjustConstructorCallParams3
- adjustConstructorCallParams4
- adjustConstructorCallParams5